PetGame/PetGame/PetData.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  PetData.swift
+//
+//  Created by Rennick Carter M. on 2019-01-31.
+//  Copyright © 2019. All rights reserved.
+//
+import Foundation
+import SpriteKit
+import GameplayKit
+
+class PetData{
+    var lastFed : NSDate
+    var lastWashed : NSDate
+    var lastWalked : NSDate
+    var lastPlayedWith : NSDate
+    
+    let hungerDuration : Double = 11.9 * 60.0 * 60.0 //Time to zero hunger in seconds
+    let washDuration : Double = 18.0 * 60.0 * 60.0   //Time to zero clean in seconds
+    let walkDuration : Double = 24.1 * 60.0 * 60.0   //Time to zero tired in seconds
+    let playDuration : Double = 48.2 * 60.0 * 60.0   //Time to zero play in seconds
+    
+    init(){
+        //Set all our values to the current date and time initially
+        lastFed = NSDate()
+        lastWashed = NSDate()
+        lastWalked = NSDate()
+        lastPlayedWith = NSDate()
+        
+        //Attempt to grab stored dates and times for when the pet was last cared for
+        let lf = UserDefaults.standard.object(forKey: "LastFed")
+        let lws = UserDefaults.standard.object(forKey: "LastWashed")
+        let lwl = UserDefaults.standard.object(forKey: "LastWalked")
+        let lpw = UserDefaults.standard.object(forKey: "LastPlayedWith")
+        
+        //If the stored data exists and can be cast to an NSDate, use that instead of current date and time
+        if let date = lf as? NSDate{
+            lastFed = date
+        }
+        
+        if let date = lws as? NSDate{
+            lastWashed = date
+        }
+        
+        if let date = lwl as? NSDate{
+            lastWalked = date
+        }
+        
+        if let date = lpw as? NSDate{
+            lastPlayedWith = date
+        }
+    }
+    
+    deinit{
+        StoreData()
+    }
+    
+    func StoreData() -> Void{
+        let defaults = UserDefaults.standard
+        defaults.set(lastFed, forKey: "LastFed")
```

Comment:
its important to drill this in, that if you used a constant, you would have the editor autocomplete for you so you dont have to worry about misspelling the string in subsequent uses.